File: C++/Arrays/buySellStock.cpp
#include<iostream>
#include<vector>
using namespace std;

int maxProfit(vector<int>& prices){
    int min = prices[0];
    int max = 0 ;
    for(int i=1;i<prices.size();i++){
        if(prices[i]<min)
            min=prices[i]; 

        if(max<prices[i]-min)
            max = prices[i]-min;
    }
    return max;
}

int main (){
    vector<int> prices = {3,2,6,5,0,3};
    cout<<maxProfit(prices);
}

File: Python/Arrays/buySellStock.py
def firstIndex(arr, n):
    # Your code goes here
    first = -1
    if n==1:
            if arr[0] == 1:
                return 0
    for i in range(0,n-1):
            if arr[i] == 1:
                first = i
    if first != -1:
        return first
    elif first == -1:
        return -1


def maxProfit(prices):
    min = prices[0]
    max = 0
    for i in range(0,len(prices)):
       if min>prices[i]:
           min = prices[i]
           
       elif max<prices[i]-min:
           max = prices[i]
    
    return max

prices = [7,6,4,3,1]
test = [1]
print(firstIndex(test,1))
# print(maxProfit(prices))